proxy/proxy.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"os"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ProxyEnvNames are standard environment variables for proxies
+var ProxyEnvNames = []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY"}
+
+// ReadProxyVarsFromEnv retrieves the standard proxy-related environment
+// variables from the running environment and returns a slice of corev1 EnvVar
+// containing upper and lower case versions of those variables.
+func ReadProxyVarsFromEnv() []corev1.EnvVar {
+	envVars := []corev1.EnvVar{}
+	for _, s := range ProxyEnvNames {
+		value, isSet := os.LookupEnv(s)
+		if isSet {
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  s,
+				Value: value,
+			})
+			// Duplicate values for upper and lower case
+			envVars = append(envVars, corev1.EnvVar{
+				Name:  strings.ToLower(s),
+				Value: value,
+			})
```

Comment:
```suggestion
			envVars = append(envVars, corev1.EnvVar{
				Name:  s,
				Value: value,
			}, corev1.EnvVar{
				Name:  strings.ToLower(s),
				Value: value,
			})
```

Reply:
is there specific reason to have lower and upper case appended? Can we always convert values to one case and add it to slice? Would it be easier to have a getter that checks and returns the specific value for the variable asked. With this we need not store two values, and user can obtain the the specific value for the key they specify. 

```go
func GetProxyValue(input string) (string, error) {
    // always convert to lower case so that we don't have to store two keys for same value
     key := strings.ToLower(input)
     if val, ok := envVars[key]; ok {
             return val, nil
     }
     return "",  fmt.Errorf("%s not set", input)
}
```
With this we may or may not want to expose `envVars` (the slice of proxy variables entirely). Its upto us. And if we want we can go a step further by making `ProxyEnvNames` to be a particular type.

If there is a specific reason to have these keys in both the cases and to return the list entirely, please feel free to ignore the comment.